apps/torch/app.js

```diff
@@ -11,14 +11,16 @@ Bangle.setLCDBrightness(1);
 Bangle.setLCDPower(1);
 Bangle.setLCDTimeout(0);
 g.reset();
+bgBackup = g.theme.bg;
 g.setTheme({bg:settings.bg,fg:"#000"});
 g.setColor(settings.bg);
 g.fillRect(0,0,g.getWidth(),g.getHeight());
 Bangle.setUI({
   mode : 'custom',
-  back : load, // B2: Clicking the hardware button or pressing upper left corner turns off (where red back button would be)
-  btn : (n)=>{ // B1: Any button turns off
-    if (process.env.HWVERSION==1 && (n==1 || n==2 || n==3)) {
+  back : ()=>{g.setTheme({bg:bgBackup});load();}, // B2: SW back button
+  btn : (n)=>{ // B1&2: Any HW button turns off
+    if (n==1 || n==2 || n==3) {
```

Comment:
Again, do we need this? I think n can ONLY be 1/2/3 so this check seems redundant?

Reply:
Since I don't have a B1 I'm not sure. But I assumed presses of the screen would fire the `btn`-function with `n==4 `or `n==5`. To me it made more sense to only have the HW buttons exit the torch and not the screen presses, and that's also how it was coded before.

Reply:
Reading this on the HW reference indicates you are right though:
 > btn : function(n) {}, // optional - handler for 'button' events (n==1 on Bangle.js 2, n==1/2/3 depending on button for Bangle.js 1)

I guess I got confused by the `n`-variable in:
>touch : function(n,e) {}, // optional - handler for 'touch' events
Where I think I've read `n` can be `n==4` or `n==5`